atomisticparsers/gromacs/parser.py

```diff
@@ -1629,6 +1682,76 @@ def parse_input(self):
         )
         sec_control_parameters.m_set(quantity_def, input_parameters)
 
+    def find_trajectory_files(self):
+        """
+        Find and set trajectory files following the priority:
+        "trr" > "xtc", fall back "pdb" > "gro".
+        Prioritize files with exact basename match over files with partial basename match over other files.
+        """
+        traj_priority_list = ['trr', 'xtc', 'pdb', 'gro']
+
+        def is_readable(file_path):
+            positions = None
+            try:
+                test_universe = MDAnalysis.Universe(
+                    self.traj_parser.mainfile, file_path
+                )
+                atoms = getattr(test_universe, 'atoms', None)
+                positions = getattr(atoms, 'positions', None)
+            except Exception:
+                self.logger.warning(
+                    'Error reading positions from trajectory file: {}'.format(file_path)
+                )
+            if positions is not None:
+                # If readable, the correct trajectory has been found
+                return True
+
+            return False
+
+        def sort_by_priority(files_dicts):
```

Comment:
I have the feeling that this function might be simplified but I need to walk through it with you when we meet

Reply:
I think I have a slightly more streamlined approach, but my main question is what is our approach for ordering type `2` files with no basename match if there are multiple options and also multiple extensions?